modules/distributor/distributor.go

```diff
@@ -398,14 +407,36 @@ func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string
 			return err
 		}
 
-		_, err = c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
+		response, err := c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
+		// response = {empty}, err = nil ==> entire batch processed successfully
+		// response = nil, err != nill ==> entire batch failed not related to livetraces or tracetoolong error
+		// response != {empty}, err = nil  ==>  at least one trace was discarded
 		metricIngesterAppends.WithLabelValues(ingester.Addr).Inc()
+
+		r.mu.Lock()
+		defer r.mu.Unlock()
+		r.responses = append(r.responses, response)
```

Comment:
i think we should check for err != nil before the locking. if err != nil there's no reason to accumulate the responses.